### ISSUE TITLE
[JENKINS-56109] Jenkins Core: Change Jenkins configuration UI from tables to divs

### ISSUE THREAD

**Issue by timja-bot** [timja-org/jenkins-gh-issues-poc-11-04#4626]
<h1><a name="Reportallissueswithacompletepluginlist%28reportingthisasanissueinthechangelogisn%27thelpinganyone%29"></a>Report all issues with a complete plugin list (reporting this as an issue in the changelog isn't helping anyone)</h1>

<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">
JENKINS_HOST=username:password@myhost.com:port
curl -sSL <span class="code-quote">"http:<span class="code-comment">//$JENKINS_HOST/pluginManager/api/xml?depth=1&xpath=<span class="code-comment">/*/*/</span>shortName|<span class="code-comment">/*/*/</span>version&wrapper=plugins"</span> | perl -pe <span class="code-quote">'s/.*?<shortName>([\w-]+).*?<version>([^<]+)()(<\/\w+>)+/\1 \2\n/g'</span>|sed <span class="code-quote">'s/ /:/'</span>
</span> </pre>
</div></div>

<h2><a name="Checkifyourpluginalreadyhasanissuereportedhttps%3A%2F%2Fissues.jenkins.io%2Fissues%2F%3Ffilter%3D22840%28artifactorypluginishttps%3A%2F%2Fgithub.com%2Fjfrog%2Fjenkinsartifactoryplugin%2Fissues%2F350"></a>Check if your plugin already has an issue reported <a href="https://issues.jenkins.io/issues/?filter=22840" class="external-link" rel="nofollow">https://issues.jenkins.io/issues/?filter=22840</a> (artifactory plugin is <a href="https://github.com/jfrog/jenkins-artifactory-plugin/issues/350" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jfrog/jenkins-artifactory-plugin/issues/350</a></h2>

<h2><a name="Ifyouknowwhichpluginiscausingtheissuereportanewissue%2Cincludingstepstoreproduceandscreenshotsandlabelitwith%27tablestodivsregression%27"></a>If you know which plugin is causing the issue report a new issue, including steps to reproduce and screenshots and label it with 'tables-to-divs-regression'</h2>

<p> </p>

<p>Historically, jenkins's jelly framework generated lots of tables. These tables are very mobile hostile. The ui it generates is also not in line with modern UIs.</p>

<p> </p>

<p>Here's one example from Windows 10's settings:</p>

<p><span class="image-wrap" style=""><img src="https://issues.jenkins.io/secure/attachment/46006/46006_image-2019-02-12-15-44-44-739.png" height="278" width="228" style="border: 0px solid black" /></span></p>

<p>Note that the label for a field is above the field – not to the left of the field.</p>

<p>Here's a much older UI from Windows:</p>

<p><span class="image-wrap" style=""><img src="https://issues.jenkins.io/secure/attachment/46005/46005_image-2019-02-12-15-45-51-351.png" height="306" width="301" style="border: 0px solid black" /></span></p>

<p>Note that the labels for checkboxes/radios are to the right of the buttons, not to their left (* with RTL layouts, this changes, but that's due to mirroring and applies to almost everything).</p>

<p> </p>

<p>Even so, labels for other things are above, not to their left (see this example for invitation duration – which is also pre Windows 10):</p>

<p><span class="image-wrap" style=""><img src="https://issues.jenkins.io/secure/attachment/46004/46004_image-2019-02-12-15-46-51-006.png" height="232" width="333" style="border: 0px solid black" /></span></p>

<p> </p>

<p>Examples can be seen in macOS, Maemo / MeeGo, and pretty much every system (including Chrome / Firefox / Android).</p>

<p> </p>

<p>This issue tracked the core jelly changes made to switch from generating tables to generating divs.</p>

---
<details><summary><i>Originally reported by <a title="Josh Soref" href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jsoref">jsoref</a>, imported from: <a href="https://issues.jenkins.io/browse/JENKINS-56109" target="_blank">Jenkins Core: Change Jenkins configuration UI from tables to divs</a></i></summary>
<i><ul>
<li><b>assignee</b>: <a title="Josh Soref" href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jsoref">jsoref</a>
<li><b>status</b>: Closed
<li><b>priority</b>: Major
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2020-10-24T23:49:53+00:00
<li><b>imported</b>: 2025-11-05
</ul></i>
</details>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=timja">timja</a>:</i>
<p>To those reporting issues against this epic in the changelog, it's not useful, please report a new issue with a full list of installed plugins as this is plugin specific</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=JIRAUSER135290">JIRAUSER135290</a>:</i>
<p>This has broken default build with parameters- Reference - <a href="https://issues.jenkins.io/browse/JENKINS-66933" class="external-link" rel="nofollow">https://issues.jenkins.io/browse/JENKINS-66933</a> . Kindly provide a solution for this or revert back the changes. You can't break one to make another thing working.</p>
